app/src/main/java/com/yeonae/chamelezone/view/home/HomeTabFragment.kt

```diff
@@ -21,45 +23,33 @@ import com.yeonae.chamelezone.view.home.presenter.HomePresenter
 import com.yeonae.chamelezone.view.login.LoginActivity
 import com.yeonae.chamelezone.view.place.PlaceDetailActivity
 import com.yeonae.chamelezone.view.search.SearchActivity
-import kotlinx.android.synthetic.main.activity_place_detail.*
 import kotlinx.android.synthetic.main.fragment_home_tab.*
 
-class HomeTabFragment : Fragment(), HomeContract.View {
+class HomeTabFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener, HomeContract.View {
     override lateinit var presenter: HomeContract.Presenter
-    private lateinit var placeResponse: PlaceResponse
     private val placeAdapter = HomePlaceRvAdapter()
-    private var memberNumber: Int? = null
+    private var memberNumber: Int? = 0
     private var placeNumber = 0
 
     override fun showHomeList(place: List<PlaceResponse>) {
         placeAdapter.addData(place)
 
-        if (::placeResponse.isInitialized) {
-            if (placeResponse.likeStatus) {
-                btn_like.isChecked = true
-            }
-        }
-
         placeAdapter.setLikeButtonListener(object : HomePlaceRvAdapter.LikeButtonListener {
-            override fun onLikeClick(placeResponse: PlaceResponse) {
-                Log.d("HomeTabFragment like memberNumber", placeResponse.memberNumber.toString())
+            override fun onLikeClick(placeResponse: PlaceResponse, isChecked: Boolean) {
+                placeNumber = placeResponse.placeNumber
+
                 if (memberNumber == null) {
-                    btn_like.setOnClickListener {
-                        btn_like.isChecked = false
-                        val intent = Intent(context, LoginActivity::class.java)
-                        startActivity(intent)
-                    }
+                    val intent = Intent(context, LoginActivity::class.java)
+                    startActivity(intent)
                 } else {
-                    btn_like.setOnClickListener {
-                        if (btn_like.isChecked) {
-                            memberNumber?.let { it1 -> presenter.selectLike(it1, placeNumber) }
-                        } else {
-                            memberNumber?.let { it1 ->
-                                presenter.deleteLike(
-                                    it1,
-                                    placeNumber
-                                )
-                            }
+                    if (isChecked) {
+                        memberNumber?.let { it1 -> presenter.selectLike(it1, placeNumber) }
```

Comment:
```suggestion
                        presenter.selectLike(memberNumber, placeNumber) }
```
if(memberNumber == null) 의 else는 not null 아닐까요?

Reply:
?.let{}을 사용하면 if(type != null) 이렇게 된다고 알고 있습니다
else가 not null 이기 때문에 null을 배제하기 위해서 사용했습니다

Reply:
다시 수정했습니다
9fa0291